### PR TITLE
Add an option to test Azure File CSI Driver

### PR DESCRIPTION
**Pull request by chewong**
/assign @andyzhangx @feiskyer 
/cc @ritazh 
/area provider/azure

### CODE REVIEW

File: config/jobs/kubernetes-sigs/azurefile-csi-driver/azurefile-csi-driver-config.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        - "--acsengine-admin-username=azureuser"
+        - "--acsengine-creds=$AZURE_CREDENTIALS"
+        - "--acsengine-orchestratorRelease=1.16"
+        - "--acsengine-mastervmsize=Standard_DS2_v2"
+        - "--acsengine-agentvmsize=Standard_D4s_v3"
+        - "--acsengine-location=eastus2"
+        - "--acsengine-public-key=$AZURE_SSH_PUBLIC_KEY_FILE"
+        - "--acsengine-template-url=https://raw.githubusercontent.com/kubernetes/cloud-provider-azure/master/tests/k8s-azure/manifest/linux.json"
+        - "--acsengine-download-url=https://github.com/Azure/aks-engine/releases/download/v0.39.1/aks-engine-v0.39.1-linux-amd64.tar.gz"
+        - "--test-azure-file-csi-driver=True"
```
Comment (ritazh) [inline config/jobs/kubernetes-sigs/azurefile-csi-driver/azurefile-csi-driver-config.yaml]: This `test-azure-file-csi-driver` flag won't take into effect until after a new `gcr.io/k8s-testimages/kubekins-e2e` image has been released with the latest kubetest code. So you might need to create separate PRs for this.

Comment (chewong) [reply config/jobs/kubernetes-sigs/azurefile-csi-driver/azurefile-csi-driver-config.yaml]: Removed the job configuration for now. I will create a separate PR for that after this is merged.

File: kubetest/azure.go [hunk 2]
```diff
@@ -925,10 +927,18 @@ func (c Cluster) TestSetup() error {
 		if err := os.Setenv("K8S_AZURE_LOCATION", c.location); err != nil {
 			return err
 		}
+	} else if *testAzureFileCSIDriver == true {
+		// Set env vars required by Azure File CSI driver tests.
+		// tenantId, subscriptionId, aadClientId, and aadClientSecret will be obtained from AZURE_CREDENTIAL
+		if err := os.Setenv("resourceGroup", c.resourceGroup); err != nil {
```
Comment (feiskyer) [inline kubetest/azure.go]: could we rename the env name for resourceGroup and location? They're usually in capital letter.

File: kubetest/azure.go [hunk 3]
```diff
@@ -992,3 +1004,38 @@ func (t *GinkgoCustomTester) Run(control *process.Control, testArgs []string) er
...
+func (t *GinkgoAzureFilCSIDriverTester) Run(control *process.Control, testArgs []string) error {
+	cmd := exec.Command("make", "e2e-test")
+	projectPath := githubProjectPath("kubernetes-sigs", "azurefile-csi-driver")
+	cmd.Dir = projectPath
+	testErr := control.FinishRunning(cmd)
+	return testErr
+}
+
+// githubProjectPath returns $GOPATH/src/github.com/<username>/<projectName>
+func githubProjectPath(username, projectName string) string {
```
Comment (feiskyer) [inline kubetest/azure.go]: could we move this to util with name `K8sigs()`?

**Pull request merged**